frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -523,6 +541,11 @@ pub mod pallet {
             Self::update_staker_info(&staker, &contract_id, staker_info);
             ContractEraStake::<T>::insert(&contract_id, current_era, staking_info);
 
+            // Insert the delegate to this contract if not yet
+            if !Delegates::<T>::contains_key(&staker, &contract_id) {
+                Delegates::<T>::insert(&staker, &contract_id, staker.clone());
+            }
```

Comment:
How can this be solved more efficiently?

Reply:
This way is much better:

```rust
Delegates::<T>::mutate(&staker, &contract_id, |delegator| {
    if delegator.is_none() {
        *delegator = Some(staker.clone());
    }
});
```